burnman/mineral.py

```diff
@@ -64,7 +65,17 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
-            warnings.warn('Overriding database equation of state. From ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__, stacklevel=2)
+
+            # Warn user that they are changing the EoS
+            warnings.warn('Overriding database equation of state from ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__ + '. This does not overwrite any mineral attributes other than temperature and pressure. Non-implemented attributes of the old method will remain unchanged unless they are explicitly changed by the user.', stacklevel=2)
+
+            # Set new EoS
+            self.params['equation_of_state']=equation_of_state
```

Comment:
I don't quite get it, you want this so that the error message reflects the old method that was used? I don't think that makes sense. We want to warn if the user sets a method that is not the one for which the parameters got derived for (and this is tracked in params[eos]). So I vote against changing this.